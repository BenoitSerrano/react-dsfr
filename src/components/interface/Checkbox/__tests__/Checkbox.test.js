import { render, screen } from '@testing-library/react';
import { Checkbox } from '../index';

describe('<Checkbox />', () => {
  it('in document', () => {
    render(
      <Checkbox
        label="Label de la checkbox"
        value="value"
      />,
    );
    const search = screen.getByTestId('checkbox-testid');
    expect(search).toBeInTheDocument();
  });

  // test de la prÃ©sence de la class rf-checkbox-group
  it('has rf-checkbox-group class', () => {
    const { container } = render(
      <Checkbox
        label="Label de la checkbox"
        value="value"
      />,
    );
    expect(container.getElementsByClassName('rf-checkbox-group').length).toBe(1);
  });

  // test de la class d'erreur
  it('has rf-select class', () => {
    const { container } = render(
      <Checkbox
        label="Label de la checkbox"
        value="value"
        messageType="error"
      />,
    );
    expect(container.getElementsByClassName('rf-checkbox-group--error').length).toBe(1);
  });
});
