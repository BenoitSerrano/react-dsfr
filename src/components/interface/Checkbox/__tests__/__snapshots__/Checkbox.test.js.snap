// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Checkbox /> renders correctly 1`] = `
<div
  className="rf-checkbox-group"
>
  <input
    data-testid="checkbox-testid"
    disabled={false}
    id={null}
    name="checkbox"
    onChange={[Function]}
    type="checkbox"
    value="value"
  />
  <label
    className="rf-label"
    htmlFor={null}
  >
    Label de la checkbox
  </label>
  
  
</div>
`;
